### PR TITLE
Adds validator to check simple products - #23182

### PR DESCRIPTION
**Pull request by Martinsl**
### Description (*)

Adds validator to check if all the simple products in the Quote are still assigned to a parent product

### Fixed Issues (if relevant)

1. magento/magento2#23182: Configurable - removed child from configurable - checkout still possible

### Manual testing scenarios (*)

1. (front) Add Configurable product to the cart
2. (front) Move to checkout
3. (admin) REMOVE configurable child and save configurable product. Of course simple child still exists in the store- but its not visible individually. BUT its not a part of configurable anymore.
4. (front) Finish checkout. An error should be displayed.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Quote/Model/ValidationRules/ConfigurableProductsValidationRule.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        $validationErrors = [];
+        $cartAllowedChildren = [];
+        $cartAllSimpleWithConfigurableParent = [];
+
+        foreach ($quote->getItemsCollection() as $item) {
+            $product = $item->getProduct();
+
+            if (!$item->isDeleted()
+                && $product->getTypeId() === 'configurable'
```
Comment (sidolov) [inline app/code/Magento/Quote/Model/ValidationRules/ConfigurableProductsValidationRule.php:62]: Module Quote shouldn't know anything about Configurable products. By adding such references we creating the coupling between modules. Please, move validator to the Configurable product module.

**Review by sidolov — CHANGES_REQUESTED** [review]
Please, cover changes with automated tests.

**Comment by lenaorobei** [comment]
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it. Thanks.

File: app/code/Magento/Quote/Model/ValidationRules/ConfigurableProductsValidationRule.php (hunk 1 above)
Comment (kandy) [reply app/code/Magento/Quote/Model/ValidationRules/ConfigurableProductsValidationRule.php:62]: Maybe even make sense to create new module QuotoForConfigurableProduct

**Pull request closed without merging**